v3/TreeJax/ts/semantic_visitor.ts

```diff
@@ -0,0 +1,510 @@
+    let attributes = node.getAttributes();
+    let sepValue = this.getAttributeDefault(attributes, 'separators', ',');
+    let open = this.getAttributeDefault(attributes, 'open', '(');
+    let close = this.getAttributeDefault(attributes, 'close', ')');
+    let semNode = sem.Processor.mfenced(open, close, sepValue, children);
+    this.removeLevel();
+    this.appendChild(semNode);
+  }
+
+  /**
+   * @override
+   */
+  protected visitNodeMenclose(node: nf.NodeMenclose) {
+    this.addLevel();
+    super.visitNodeMenclose(node);
+    let children = this.levels.get(this.level);
+    let semNode = this.factory.makeBranchNode(
+      sem.Type.ENCLOSE, [sem.Processor.row(children)], []);
+    semNode.role = node.getAttributes()['notation'] || sem.Role.UNKNOWN;
+    this.removeLevel();
+    this.appendChild(semNode);
+  }
+
+  /**
+   * @override
+   */
+  protected visitNodeMsub(node: nf.NodeMsub) {
+    this.limitNode('MSUB', node, super.visitNodeMsub.bind(this));
+  }
+
+  /**
+   * @override
+   */
+  protected visitNodeMsup(node: nf.NodeMsup) {
+    this.limitNode('MSUP', node, super.visitNodeMsup.bind(this));
+  }
+
+  /**
+   * @override
+   */
+  protected visitNodeMsubsup(node: nf.NodeMsubsup) {
+    this.limitNode('MSUBSUP', node, super.visitNodeMsubsup.bind(this));
+  }
+
+  /**
+   * @override
+   */
+  protected visitNodeMunder(node: nf.NodeMunder) {
+    this.limitNode('MUNDER', node, super.visitNodeMunder.bind(this));
+  }
+
+  /**
+   * @override
+   */
+  protected visitNodeMover(node: nf.NodeMover) {
+    this.limitNode('MOVER', node, super.visitNodeMover.bind(this));
+  }
+
+  /**
+   * @override
+   */
+  protected visitNodeMunderover(node: nf.NodeMunderover) {
+    this.limitNode('MUNDEROVER', node, super.visitNodeMunderover.bind(this));
+  }
+
+  // /**
+  //  * @override
+  //  */
+  // protected visitNodeMmultiscripts(node: nf.NodeMmultiscripts) {
+  //   super.visitNodeMmultiscripts(node);
+  // }
+
+  // /**
+  //  * @override
+  //  */
+  // protected visitNodeMprescripts(node: nf.NodeMprescripts) {
+  //   super.visitNodeMprescripts(node);
+  // }
+
+  /**
+   * @override
+   */
+  protected visitNodeMtable(node: nf.NodeMtable) {
+    this.addLevel();
+    super.visitNodeMtable(node);
+    let children = this.levels.get(this.level);
+    let semNode = this.factory.makeBranchNode(sem.Type.TABLE, children, []);
+    sem.Processor.tableToMultiline(semNode);
+    this.removeLevel();
+    this.appendChild(semNode);
+  }
+
+  // /**
+  //  * @override
+  //  */
+  // protected visitNodeMlabeledtr(node: nf.NodeMlabeledtr) {
+  //   super.visitNodeMlabeledtr(node);
+  // }
+
+  /**
+   * @override
+   */
+  protected visitNodeMtr(node: nf.NodeMtr) {
+    this.addLevel();
+    super.visitNodeMtr(node);
+    let children = this.levels.get(this.level);
+    let semNode = this.factory.makeBranchNode(sem.Type.ROW, children, []);
+    semNode.role = sem.Role.TABLE;
+    this.removeLevel();
+    this.appendChild(semNode);
+  }
+
```

Comment:
Try to combine all the elements creating branch nodes  into a "macro" method, to reduce code duplication.